src/sentry/tagstore/models.py

```diff
@@ -11,23 +11,46 @@
 from django.conf import settings
 from django.core.exceptions import ImproperlyConfigured
 
-# HACK: This was taken from nodestore.models. Django doesn't play well with our
-# naming schemes, and we prefer our methods ways over Django's limited scoping
-if settings.SENTRY_TAGSTORE.startswith('sentry.tagstore.legacy'):
-    from sentry.tagstore.legacy.models import *  # NOQA
-elif settings.SENTRY_TAGSTORE.startswith('sentry.tagstore.v2'):
-    from sentry.tagstore.v2.models import *  # NOQA
-
-    # Prevent schema migration from adding DROP TABLE for Legacy
-    from sentry.tagstore.legacy import models as _legacy_models  # NOQA
+backends = []
+
+if settings.SENTRY_TAGSTORE == 'sentry.utils.services.ServiceDelegator':
+    backends = [
+        backend: backend['path'] for backend in
+        settings.SENTRY_TAGSTORE_OPTIONS.get('backends', {}).values()
+    ]
 elif settings.SENTRY_TAGSTORE.startswith('sentry.tagstore.multi'):
-    for backend in settings.SENTRY_TAGSTORE_OPTIONS.get('backends', []):
-        if backend[0].startswith('sentry.tagstore.legacy'):
-            from sentry.tagstore.legacy.models import *  # NOQA
-        elif backend[0].startswith('sentry.tagstore.v2'):
-            from sentry.tagstore.v2 import models as _v2_models  # NOQA
```

Comment:
This import order is now non-deterministic, if we want to prioritize one over the other we can sort them by some arbitrary score.